### PR TITLE
i2c: i2c_ll_stm32_v2: fix NACK handling for STM32L4

### PR DESCRIPTION
**Pull request by vaussard**
On STM32L4, a NACK on the I2C bus triggers an I2C event interrupt (see
for example Figure 420 in the STM32L4x6 TRM). But the current logic
expects an I2C error interrupt. As a result, the NACK interrupt is
not acknowledged and leads to an IRQ storm which freezes the system.

Factorize the NACK handling code to share it between event and error
ISRs, then call it from the event ISR if we compile for STM32L4.

### CODE REVIEW

**Review by ydamigos — CHANGES_REQUESTED** [review]
Actually, NACK is an I2C event interrupt for all series supported by i2c_ll_stm32_v2. We just need to move NACK handling code to event ISR.

**Comment by vaussard** [comment]
> Actually, NACK is an I2C event interrupt for all series supported by i2c_ll_stm32_v2. We just need to move NACK handling code to event ISR.

Indeed, this is the case on STM32F0/F3/L0/L4. I moved the NACK handling to the event ISR and updated the commit message accordingly.

File: drivers/i2c/i2c_ll_stm32_v2.c [hunk 1]
```diff
@@ -117,7 +117,14 @@ void stm32_i2c_event_isr(void *arg)
...
 	LL_I2C_DisableIT_TX(i2c);
 	LL_I2C_DisableIT_RX(i2c);
-	data->current.is_err = 1;
+
+	if (LL_I2C_IsActiveFlag_NACK(i2c)) {
+		LL_I2C_ClearFlag_NACK(i2c);
+		data->current.is_nack = 1;
+	} else {
+		data->current.is_err = 1;
+	}
```
Comment (superna9999) [inline drivers/i2c/i2c_ll_stm32_v2.c]: This does not seem to be in the right place in the event handler, it should be handled before looking at the message it self.

Comment (superna9999) [reply drivers/i2c/i2c_ll_stm32_v2.c]: @vaussard Here is the Linux implementation of the same IP (STM32F7 uses the same i2c IP) https://github.com/torvalds/linux/blob/master/drivers/i2c/busses/i2c-stm32f7.c#L670

**Comment by aurel32** [comment]
@vaussard , this bug has been fixed as part of PR #5224, more precisely in commit 81cfdec3a. I confirm it works fine with an STM32F7 SoC. If you can confirm the issue is also fixed in your case, I guess this PR can just be closed.

**Comment by vaussard** [comment]
Indeed, commit 81cfdec should also fixe the issue. Closing this PR, I did not had a lot of time to work on it anyway.

**Pull request closed without merging**